_posts/2019-04-26-using-idea-with-unetstack.md

```diff
@@ -0,0 +1,171 @@
+
+`sudo snap install intellij-idea-community --classic`
+
+Following this, download the latest copy of UnetStack for your operating system from [here](https://www.unetstack.net/downloads.html). Extract the files in a directory of your choice.
+
+From here, we can look at how we can create a new `UnetAgent` using IntelliJ.
+
+## Setting up your project
+
+On the first run of IDEA, you will be presented with the following Welcome Screen:
+
+![](assets/img/idea-setup/0.png)
+
+Click on create a new project and choose "Java" project on the following screen. Leave the Groovy and Kotlin/JVM options unselected. We will be adding our own libraries later on.
+
+![](assets/img/idea-setup/1.png)
+
+Leave the "Create project from template" option unselected on the following screen.
+
+![](assets/img/idea-setup/2.png)
+
+In the next screen you will have to choose the name of your project. Let's call it `MyAwesomeAgent` for the purpose of this tutorial. This name doesn't really need to follow any convention as we won't be using it later on.
+
+![](assets/img/idea-setup/3.png)
+
+Once you've completed all these steps, you will notice that IDEA has created a new directory with the name of your project with the following directories:
+
+```
+src/
+MyAwesomeAgent.iml
+```
+
+We need to create all of source files in the `src/` directory. Before doing so, let's add the Unet JARs to our projects.
+
+Go to __File > Project Structure__ in the toolbar. Open the __Libraries__ pane as shown in the left hand column. You will be presented with the following screen:
+
+![](assets/img/idea-setup/4.png)
+
+Select Java project library and look for the Unet JARs you extracted in the Installation step of this tutorial. The exact location depends on the version of Unet and OS you're using. For example, in `unetsim-1.4-linux`, these JARs are available in `unetsim-1.4-linux/UnetIDE/app/lib` and in the pre-release build of UnetStack3, these JARs can be found in `UnetStack3/lib`. Either way, you will only need to select the top-level directory where all the JARs are present. IDEA will automatically add all the JARs in this directory to your project.
+
+![](assets/img/idea-setup/1.png)
+
+After doing this, your Project settings screen should look like this:
+
+![](assets/img/idea-setup/6.png)
+
+We can now start writing our agent.
+
+## Writing the Agent
+
+Create a new Groovy in the `src/` directory of your project by right-clicking the `src` directory in Project pane in IDEA.
+
+![](assets/img/idea-setup/project.png)
+
+In this example, the agent we're writing will be called `AwesomeAgent`. The code for the same is below. For more information about writing agents, refer to the Unet [documentation](https://www.unetstack.net/unet-agents.html).
+
+```
```

Comment:
Can we add language hint to this (and all other) fenced code block?? I believe our markdown renderer will use that for automatic syntax highlighting. https://jekyllrb.com/docs/configuration/markdown/

Would look something like this

```
```java
```

Reply:
Thanks! I have added this hint.